torch_dicom/preprocessing/crop.py

```diff
@@ -237,3 +204,209 @@ def unapply_to_coords(coords: Tensor, bounds: Tensor, clip: bool = True) -> Tens
+
+    Args:
+        img_key: Key for the image in the example dict.
+
+        bounds_key: Key for the crop bounds in the example dict. The determined
+            crop bounds will be stored in this location.
+
+        img_dest_key: Key for the cropped image in the example dict. If None,
+            the cropped image will be stored in the same location as the original image.
+
+    Shape:
+        - Input: :math:`(N, *, H, W)`
+        - Output: :math:`(N, *, H', W')`
+    """
+
+    def __call__(self, example: E) -> E:
+        """
+        Applies the MinMaxCrop operation to the given example.
+
+        Args:
+            example: The example to be processed.
+
+        Returns:
+            The processed example.
+        """
+        img = example[self.img_key]
+        bounds = self.get_bounds(img)
+        return super().__call__(example, bounds)
+
+    @staticmethod
+    @torch.no_grad()
+    def get_bounds(x: Tensor) -> Tensor:
+        r"""Returns the crop bounds for the given image.
+
+        Args:
+            x: Image tensor.
+
+        Shape:
+            - Input: :math:`(N, *, H, W)`
+            - Output: :math:`(N, 4)`
+
+        Returns:
+            Absolute crop bounds in the format :math:`(x_1, y_1, x_2, y_2)`.
+        """
+        # Validate / process shape
+        if x.ndim < 3:
+            raise ValueError(f"Expected input to have at least 3 dimensions, got {x.ndim}")
+        elif x.ndim > 3:
+            # Reduce * dimension by taking a max
+            x = reduce(x, "b ... h w -> b h w", "max")
+        assert x.ndim == 3, f"Expected input to have 3 dimensions, got {x.ndim}"
+
+        # Create initial result where bounds are the full image
+        N, H, W = x.shape
+        bounds = torch.tensor([[0, 0, H, W]], device=x.device, dtype=torch.long)
+        bounds = repeat(bounds, "b d -> (b c) d", c=N, d=4)
+
+        # Find the first and last nonzero pixel in each row and column
+        nonzero_b, nonzero_h, nonzero_w = x.nonzero(as_tuple=True)
+
+        if nonzero_b.numel():
+            # Ignore "in beta" warning
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                # Update x1 and y1 based on minimum nonzero coordinate in each row and column
+                bounds[..., 0].scatter_reduce_(0, nonzero_b, nonzero_w, reduce="amin", include_self=False)
+                bounds[..., 1].scatter_reduce_(0, nonzero_b, nonzero_h, reduce="amin", include_self=False)
+                # Update x2 and y2 based on maximum nonzero coordinate in each row and column
+                # Add 1 to the max coordinate to get the correct crop
+                bounds[..., 2].scatter_reduce_(0, nonzero_b, nonzero_w, reduce="amax", include_self=False).add_(1)
+                bounds[..., 3].scatter_reduce_(0, nonzero_b, nonzero_h, reduce="amax", include_self=False).add_(1)
+
+        assert bounds.shape == (N, 4), f"Expected bounds to have shape (B, 4), got {bounds.shape}"
```

Comment:
Replace assert message with this? `f"Expected bounds to have shape ({N}, 4), got {bounds.shape}"`